app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RequestActions.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.{ActionBuilder, ActionRefiner}
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.{AuthenticatedCaseRequest, AuthenticatedRequest}
+
+@Singleton
+class RequestActions @Inject()(authoriseFilter: AuthoriseFilterAction,
+                               checkPermissionsAction: CheckPermissionsAction,
+                               authenticatedAction: AuthenticatedAction,
+                               caseExistsActionFactory: VerifyCaseExistsActionFactory) {
+
+  def caseExistsAndFilterByAuthorisation(reference: String): ActionBuilder[AuthenticatedCaseRequest] = authenticatedAction andThen caseExists(reference) andThen authoriseFilter
+
+  def caseExistsAndCheckPermissions(reference: String): ActionBuilder[AuthenticatedCaseRequest] = authenticatedAction andThen caseExists(reference) andThen checkPermissionsAction
```

Comment:
When I call this method, what should I expect `checkPermissions` to do?

As above I think we can make it easier to understand by having separate action calls